src/Criteria/Criteria.php

```diff
@@ -49,10 +49,35 @@ public function toQuery(): string
         return $sql;
     }
 
+    public function toQuerySearch(): string
+    {
+        $sql = '';
+        if (!empty($this->filters)) {
+            $sql .= ' WHERE ';
+            foreach ($this->filters as $fieldName => $value) {
```

Comment:
this is ... one way.

take a look at array reduce: https://www.php.net/manual/en/function.array-reduce.php

Reply:
stai asa ca tomai m-am uitat pe tot fisierul si e cam duplicat codul asta :))

Reply:
ce zici de ceva de genul:
```
// solution 1
$sql .= implode(' AND ', array_map(function ($filterName, $filterValue) {
    return sprintf('%s LIKE %s', $filterName, $filterValue);
}, array_keys($this->filters), $this->filters));

// solution 2
$sql .= array_reduce($this->filters, function ($carry, $value, $key) {
    if ($carry !== '') {
        $carry .= ' AND ';
    }
    return sprintf('%s %s LIKE %s', $carry, $key, $value);
}, '');
```

Reply:
la a 2-a vedere, propunerile mele sunt doar o varianta mai complicata a versiunii tale.

Pe de alta parte, ai o gresala: ` $sql .= $fieldName . ' LIKE :' . $fieldName;` ar trebui sa fie ` $sql .= $fieldName . ' LIKE ' . $value;`

Reply:
aici vrea sa fie "WHERE fieldName LIKE :fieldName" si fac apoi bindValue(:fieldName, %value%), nu e ok asa?

Reply:
ba da, daca faci mai apoi bind